src/sentry/incidents/models/alert_rule.py

```diff
@@ -585,13 +601,30 @@ class Meta:
 
 
 @register_alert_subscription_callback(AlertRuleMonitorType.ACTIVATED)
-def clean_expired_alerts(subscription: QuerySubscription) -> bool:
+def update_alert_activations(
+    subscription: QuerySubscription, alert_rule: AlertRule, value: float
+) -> bool:
+    alert_rule.activations.filter(finished_at=None, query_subscription=subscription).update(
+        metric_value=value
+    )
+
+    return True
+
+
+@register_alert_subscription_callback(AlertRuleMonitorType.ACTIVATED)
+def clean_expired_alerts(
+    subscription: QuerySubscription, alert_rule: AlertRule, value: float
+) -> bool:
     now = timezone.now()
     subscription_end = subscription.date_added + timedelta(
         seconds=subscription.snuba_query.time_window
     )
 
     if now > subscription_end:
+        alert_rule.activations.filter(finished_at=None, query_subscription=subscription).update(
+            metric_value=value, finished_at=now
```

Comment:
That's a good call
I wasn't sure which trigger would fire first (i assume the update since it's registered first?)

I'll remove the metric value update from here 👍

Reply:
actually 🤔 

Let me just roll both the 'update' and the 'clean expired' into a single callback